packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -200,7 +236,10 @@ class _TextSelectionToolbarState extends State<_TextSelectionToolbar> with Ticke
                       : localizations.moreButtonTooltip,
                 ),
               ),
-              ...items,
+              ...itemDatas.map((_ItemData itemData) {
```

Comment:
nit: a collection for statement that loops from 0 to `itemDatas.length` seems a better fit here? Having side effects like `itemIndex++` is a bit weird in map.

Reply:
Yes that is much better, thanks.  I forgot about the fanciness with for loops in an array.